tests/gcm/test_unit_change.py

```diff
@@ -112,3 +118,104 @@ def test_given_fitted_nonlinnear_mechanisms_when_evaluate_unit_change_linear_met
             pd.DataFrame(data=dict(A=np.random.normal(size=100), B=np.random.normal(size=100))),
             ["A", "B"],
         )
+
+
+@flaky(max_runs=5)
+def test_given_fitted_mechanisms_with_no_input_change_when_evaluate_unit_change_input_only_methods_then_returns_zero_attributions():
+    num_rows = 100
+    A = np.random.normal(size=num_rows)
+    B = np.random.normal(size=num_rows)
+    C = 3 * A + 2 * B
+
+    background_df = pd.DataFrame(data=dict(A=A, B=B, C=C))
+    foreground_df = pd.DataFrame(data=dict(A=A, B=B, C=C))
+
+    actual_contributions = unit_change_nonlinear_input_only(
+        SklearnRegressionModel(RFR().fit(np.column_stack((A, B)), C)),
+        background_df,
+        foreground_df,
+        ["A", "B"],
+    )
+    expected_contributions = pd.DataFrame(data=dict(A=np.zeros(num_rows), B=np.zeros(num_rows)))
+    np.testing.assert_array_almost_equal(actual_contributions.to_numpy(), expected_contributions.to_numpy(), decimal=1)
+
+
+@flaky(max_runs=5)
+def test_given_fitted_linear_mechanism_with_input_change_when_evaluate_unit_change_linear_input_only_then_returns_correct_attributions():
+    num_rows = 100
+    A1 = np.random.normal(size=num_rows)
+    B1 = np.random.normal(size=num_rows)
+    C1 = 3 * A1 + 2 * B1
+
+    A2 = np.random.normal(size=num_rows)
+    B2 = np.random.normal(size=num_rows)
+    C2 = 3 * A2 + 2 * B2
+
+    background_df = pd.DataFrame(data=dict(A=A1, B=B1, C=C1))
+    foreground_df = pd.DataFrame(data=dict(A=A2, B=B2, C=C2))
+
+    fitted_linear_reg = LinearRegression()
+    fitted_linear_reg.coef_ = np.array([3, 2])
+
+    actual_contributions = unit_change_linear_input_only(
+        SklearnLinearRegressionModel(fitted_linear_reg), background_df, foreground_df, ["A", "B"]
+    )
+    expected_contributions = pd.DataFrame(data=dict(A=3 * (A2 - A1), B=2 * (B2 - B1)))
+    np.testing.assert_array_almost_equal(actual_contributions.to_numpy(), expected_contributions.to_numpy(), decimal=1)
```

Comment:
It looks like you don't really need `to_numpy()` as `assert_array_almost_equal` does this for you.

Reply:
Something I know now. Thank you. Force pushing the fix.